### ISSUE TITLE
Webhook methods does not delete webhooks properly

### ISSUE THREAD

**Issue by jani123** [n8n-io/n8n#387]
**Describe the bug**

When running active / execute worflow from UI, there is extra webhooks left behind into external system. Assuming because of checkExists method does not get workflow static data for node.

**To Reproduce**
1. Create a workflow with a trigger
2. Activate workflow from UI
3. There is webhook exists in the external system
4. Press execute workflow button from UI
5. Now there is two webhooks exists in external system
6. Stop execute workflow from UI
7. One webhook is left in external system
8. Deactivate workflow from UI
9. There is still one webhook exists in external system

**Expected behavior**

I expect that there is no active webhooks left.

**Environment (please complete the following information):**
 - OS: Linux Mint 18.2 Sonya
- Docker image: sha256:42e4f9ff04c5ec87a39562aa25a969c9b1c83204b05a6a81ab0c545f87abb79f 10h old

**Additional context**

My webhook trigger methods(pretty much copied from ActiveCampaignTrigger but changed to support own system):
```
	webhookMethods = {
		default: {
			async checkExists(this: IHookFunctions): Promise<boolean> {
				const data = this.getWorkflowStaticData('node');
                                // Bug: We never get anything to data in here
				if (data.webhookId === undefined) {
					return false;
				}
				try {
					await RequestFunction.call(this, 'GET', `/v1/webhooks/${data.webhookId}`, {});
				} catch (e) {
					return false;
				}
				return true;
			},

			async create(this: IHookFunctions): Promise<boolean> {
				const url = this.getNodeWebhookUrl('default') as string;
				const data = this.getWorkflowStaticData('node');
				const types = this.getNodeParameter('eventTypes', []) as string[];
				const body: IDataObject = {
					active: true,
					url: url,
					types: types,
				};
				const webhook = await RequestFunction.call(this, 'POST', '/v1/webhooks', body);

				data.webhookId = webhook.id;
				return true;
			},

			async delete(this: IHookFunctions): Promise<boolean> {
				const data = this.getWorkflowStaticData('node');

				if (data.webhookId === undefined) {
					return false;
				}

				try {
					await RequestFunction.call(this, 'DELETE', `/v1/webhooks/${data.webhookId}`, {});
				} catch(error) {
					return false;
				}

				delete data.webhookId;

				return true;
			},
		},
	};
```
The problem seem to be that checkExists method does not get proper webhook id from `this.getWorkflowStaticData('node');` I assume that if checkExists would be able to tell n8n core code that webhook already exists, the pressing of execute worflow would update the existing webhook and not create new one. And same that when you cancel the execution, it should update it back to the production url.

**Comment by janober** [comment]
Thanks a lot for the bug report!

I can confirm the bug. Active and testing by itself seem to work perfectly fine but when doing it at the same time something gets apparently messed up. I will have a look.

**Comment by janober** [comment]
Ok found the bug and fixed it. Will release the new version with fix in the next days.

**Comment by janober** [comment]
Got released with n8n@0.58.0
